Sources/Commands/SwiftTool.swift

```diff
@@ -878,12 +878,20 @@ public class SwiftTool {
                 cachePath = try self.getCachePath().map{ $0.appending(component: "manifests") }
             }
 
+            let extraManifestFlags: [String]
+            // Disable the implicit concurrency import if the compiler in use supports it to avoid warnings if we are building against an older SDK that does not contain a Concurrency module.
+            if SwiftTargetBuildDescription.checkSupportedFrontendFlags(flags: ["disable-implicit-concurrency-module-import"], fs: localFileSystem) {
+                extraManifestFlags = self.options.manifestFlags + ["-Xfrontend", "-disable-implicit-concurrency-module-import"]
+            } else {
+                extraManifestFlags = self.options.manifestFlags
+            }
```

Comment:
Nitpick:  If we end up adding more cases here, it would probably be better to switch to an accumulator approach, with a mutable array of flags and a sequence of cases that append to it.  Otherwise it's easy for one of the cases to accidentally leave out earlier flags.  I don't have a strong feeling about it when it's just one condition, though I do think it would be a bit cleaner even in that case — but definitely something to keep in mind if this gets any more complicated.

Reply:
Make sense to me, I'm just used to preferring the immutable variant but in this case I agree that it is actually worse. Since this is a cherry-pick, I won't change it here, but I'll submit a PR for `main`.

Reply:
https://github.com/apple/swift-package-manager/pull/3571